src/app/Http/Controllers/ApiController.php

```diff
@@ -51,10 +52,26 @@ public function locations(Request $request) {
             $locations->closeTo($lat,$lng);
         }
 
-        $locations = $locations->paginate(env('LOCATION_PAGE_SIZE', 100))->appends(compact([
-            'q',
-            'available',
-        ]));
+        /**
+         * If there is a limit parameter, limit our search based on that.
+         * If there is no limit parameter, send all results with pagination.
+         */
+        if($limit != 0) {
+            /**
+             * If the user wants more than 50 results, they can just download
+             * all the results and do more searching themselves. If we had
+             * no limit, the user could (accidentally) DoS themselves by asking
+             * for, and getting, every location in the database in a single
+             * response. That's what pagination is for!
+             */
+            $limit = $limit > 50 ? 50 : $limit;
+            $locations = $locations->limit($limit)->get();
```

Comment:
We don't want to use`limit` with `paginate` because it will prevent the pagination, I believe.

Reply:
It does. In fact, you **cannot** use them together (as I learned tonight). However, in this case because we are doing the limit and the pagination as mutually exclusive, it is not a problem. That said, the approach that you described above is better and we'll go with that.